### PR TITLE
add explain sql test for optimizer rule PreCastLitInComparisonExpressions

### PR DESCRIPTION
**Pull request by liukun4515**
# Which issue does this PR close?

add SQL level test for the https://github.com/apache/arrow-datafusion/pull/3270

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/core/tests/sql/explain_analyze.rs [hunk 1]
```diff
@@ -373,6 +373,28 @@ async fn csv_explain_verbose() {
...
+async fn csv_explain_inlist_verbose() {
+    let ctx = SessionContext::new();
+    register_aggregate_csv_by_sql(&ctx).await;
+    let sql = "EXPLAIN VERBOSE SELECT c1 FROM aggregate_test_100 where c2 in (1,2,4)";
+    let actual = execute(&ctx, sql).await;
+
+    // Optimized by PreCastLitInComparisonExpressions rule
+    // the data type of c2 is INT32, the type of `1,2,3,4` is INT64.
+    // the value of `1,2,4` will be casted to INT32 and pre-calculated
+    // flatten to a single string
```
Comment (alamb) [inline datafusion/core/tests/sql/explain_analyze.rs:386]:
```suggestion

    // flatten to a single string
```

File: datafusion/core/tests/sql/explain_analyze.rs [hunk 2]
```diff
@@ -373,6 +373,28 @@ async fn csv_explain_verbose() {
...
+    let actual = actual.into_iter().map(|r| r.join("\t")).collect::<String>();
+    assert_contains!(
+        &actual,
+        "#aggregate_test_100.c2 IN ([Int64(1), Int64(2), Int64(4)])"
+    );
+    assert_contains!(
```
Comment (alamb) [inline datafusion/core/tests/sql/explain_analyze.rs:392]:
```suggestion
    // after optimization (casted to Int32)
    assert_contains!(
```

File: datafusion/core/tests/sql/explain_analyze.rs [hunk 3]
```diff
@@ -373,6 +373,28 @@ async fn csv_explain_verbose() {
...
+    register_aggregate_csv_by_sql(&ctx).await;
+    let sql = "EXPLAIN VERBOSE SELECT c1 FROM aggregate_test_100 where c2 in (1,2,4)";
+    let actual = execute(&ctx, sql).await;
+
+    // Optimized by PreCastLitInComparisonExpressions rule
+    // the data type of c2 is INT32, the type of `1,2,3,4` is INT64.
+    // the value of `1,2,4` will be casted to INT32 and pre-calculated
+    // flatten to a single string
+    let actual = actual.into_iter().map(|r| r.join("\t")).collect::<String>();
+    assert_contains!(
```
Comment (alamb) [inline datafusion/core/tests/sql/explain_analyze.rs:388]:
```suggestion

    // before optimization (Int64 literals)
    assert_contains!(
```

**Review by alamb — APPROVED** [review]
Thanks @liukun4515

**Comment by andygrove** [comment]
test failure caused by https://github.com/apache/arrow-datafusion/issues/3327

**Comment by liukun4515** [comment]
> Thanks @liukun4515

resolve the comments manually

**Pull request merged**